subsys/bluetooth/audio/ascs.c

```diff
@@ -114,9 +138,29 @@ static void ase_status_changed(struct bt_bap_ep *ep, uint8_t old_state, uint8_t
 		bt_conn_get_info(conn, &conn_info);
 
 		if (conn_info.state == BT_CONN_STATE_CONNECTED) {
+			const uint8_t att_header_size = 3; /* opcode (1) + handle (2) */
+			const uint16_t max_size = bt_gatt_get_mtu(conn) - att_header_size;
+			uint16_t ntf_size;
+			int err;
+
+			err = k_sem_take(&ase_buf_sem, ASE_BUF_SEM_TIMEOUT);
+			if (err != 0) {
+				LOG_DBG("Failed to take ase_buf_sem: %d", err);
+
+				return;
+			}
+
 			ascs_ep_get_status(ep, &ase_buf);
 
-			bt_gatt_notify(conn, ase->attr, ase_buf.data, ase_buf.len);
+			ntf_size = MIN(max_size, ase_buf.len);
```

Comment:
Would be nice for factor out to some helper function
```suggestion
			ntf_size = get_ntf_max_size(ase_buf.len);
```

Reply:
Hmm, this one I am less sure about. It's a simple calculation which is only done once. 

Adding a helper function for this would be equally valid as doing an helper function for `max_ntf_size = bt_gatt_get_mtu(conn) - att_ntf_header_size;` or any other expression with 2 operands. 

I can add it if you really think it will be useful, otherwise suggest to just leave it as is.